app/models/user.rb

```diff
@@ -0,0 +1,27 @@
+class User < ApplicationRecord
+  has_secure_password
+
+  def mark_as_authenticated
+    @authenticated = true
+  end
+
+  def authenticated?
+    @authenticated
+  end
+
+  Guest = Naught.build do |config|
```

Comment:
Note: we are using the `naught` gem from Avdi Grimm to implement the "Null Object" pattern. This allows for smarter "nil" values that act like the model objects they are replacing, but with reasonable default behavior (such as defaulting `authenticated?` to `false`, or setting a "Honored Guest" user name)